### PR TITLE
Update Gitea to v1.21.4

### PR DESCRIPTION
**Pull request by wkobiela**
## Description

This PR includes the following changes:

1. Update Gitea to v1.21.4
2. Use redesigned shared folder structure
3. Amend upgrade and install wizards

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Package update

### CODE REVIEW

**Comment by wkobiela** [comment]
```
Gitea requires Go 1.21 or greater to build. You can get it at https://go.dev/dl/
```

**Comment by efussi** [comment]
@wkobiela what about updating Gitea to v1.20.5 which doesn't require a Go version update (as an intermediate step), so we don't have to wait for https://github.com/cloudflare/cloudflared/issues/1054?

**Comment by wkobiela** [comment]
Trying to do that here https://github.com/SynoCommunity/spksrc/pull/5931

**Comment by wkobiela** [comment]
As go was updated, trying to bump gitea further over 1.21.0 release

older PR with newest bump < 1.21.0 https://github.com/SynoCommunity/spksrc/pull/5931

**Comment by wkobiela** [comment]
Tested on DSM7.1 x64
fresh install with mysql database working OK
update from 1.17.3 with mysql database -> to 1.21.4 working OK

**Comment by wkobiela** [comment]
Is there anything more I can do to merge this package?

**Comment by hgy59** [comment]
@wkobiela can you please update the check list in the description? So far no item is checked.

**Comment by wkobiela** [comment]
Updated. Tested on vm dsm7, don't have an option to test on dsm6 right now

File: spk/gitea/src/wizard/install_uifile [hunk 1]
```diff
@@ -1,58 +1,33 @@
...
                                 "expr": "/^[\\w _-]+$/",
-                                "errorText": "Subdirectories are not supported."
+                                "errorText": "Subdirectories are not supported."              
                             }
                         }
                     }
                 ]
-            }
+                }, {
+                    "desc": ""
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile:23]:
```suggestion
```

File: spk/gitea/src/wizard/install_uifile [hunk 2]
```diff
@@ -1,58 +1,33 @@
...
+                }, {
+                    "desc": "If you let the installer create the shared folder, it is created under the same volume as the package is installed.<br/>If you want to use a different volume for the share, you must create the shared folder in DSM Control Panel before, and enter the name of the existing share in the field above."
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile:25]:
```suggestion
                    "desc": "If you let the installer create the shared folder, it is created under the same volume as the package is installed. If you want to use a different volume for the share, you must create the shared folder in DSM Control Panel before, and enter the name of the existing share in the field above."
```

File: spk/gitea/src/wizard/install_uifile [hunk 3]
```diff
@@ -1,58 +1,33 @@
...
+                }, {
+                    "desc": ""
+                }, {
+                    "desc": "This package runs as internal service user <b>'sc-gitea'</b> in DSM. The shared folder above is configured at installation time to be accessible by this user.<p>Please read <a target=\"_blank\" href=\"https://github.com/SynoCommunity/spksrc/wiki/Permission-Management\">Permission Management</a> for details."
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile:29]:
```suggestion
                    "desc": "This package runs as internal service user <b>'sc-gitea'</b> in DSM. The shared folder above is configured at installation time to be accessible by this user. Please read <a target=\"_blank\" href=\"https://github.com/SynoCommunity/spksrc/wiki/Permission-Management\">Permission Management</a> for details."
```

File: spk/gitea/src/wizard/install_uifile_plk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                            "allowBlank": false,
+                            "regex": {
+                                "expr": "/^[\\w _-]+$/",
+                                "errorText": "Podkatalogi nie są obsługiwane."              
+                            }
+                        }
+                    }
+                ]
+                }, {
+                    "desc": ""
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile_plk:23]:
```suggestion
```

File: spk/gitea/src/wizard/install_uifile_plk [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                }, {
+                    "desc": "Jeśli zezwolisz instalatorowi na utworzenie folderu współdzielonego, zostanie on utworzony na tym samym wolumenie, na którym instalowany jest pakiet.<br/>Jeśli chcesz użyć innego wolumenu, musisz wcześniej utworzyć folder współdzielony w Panelu Sterowania DSM i wpisać nazwę istniejącego folderu współdzielonego w powyższym polu."
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile_plk:25]:
```suggestion
                    "desc": "Jeśli zezwolisz instalatorowi na utworzenie folderu współdzielonego, zostanie on utworzony na tym samym wolumenie, na którym instalowany jest pakiet. Jeśli chcesz użyć innego wolumenu, musisz wcześniej utworzyć folder współdzielony w Panelu Sterowania DSM i wpisać nazwę istniejącego folderu współdzielonego w powyższym polu."
```

File: spk/gitea/src/wizard/install_uifile_plk [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                }, {
+                    "desc": ""
+                }, {
+                    "desc": "Ten pakiet uruchamia się jako użytkownik usługi wewnętrznej <b>'sc-gitea'</b> w DSM. Powyższy folder współdzielony jest skonfigurowany podczas instalacji tak, aby był dostępny dla tego użytkownika.<p>Przeczytaj <a target=\"_blank\" href=\"https://github.com/SynoCommunity/spksrc/wiki/Permission-Management\">Zarządzanie uprawnieniami</a>, aby uzyskać szczegółowe informacje."
```
Comment (mreid-tt) [inline spk/gitea/src/wizard/install_uifile_plk:29]:
```suggestion
                    "desc": "Ten pakiet uruchamia się jako użytkownik usługi wewnętrznej <b>'sc-gitea'</b> w DSM. Powyższy folder współdzielony jest skonfigurowany podczas instalacji tak, aby był dostępny dla tego użytkownika. Przeczytaj <a target=\"_blank\" href=\"https://github.com/SynoCommunity/spksrc/wiki/Permission-Management\">Zarządzanie uprawnieniami</a>, aby uzyskać szczegółowe informacje."
```

**Comment by mreid-tt** [comment]
@wkobiela I've tested under DSM 6 and it seems to work okay. I see your note for upgrades to earlier versions and I had to manually edit my config using:
```
sudo vi /var/packages/gitea/var/conf.ini
```
This was to treat with the following warnings in the admin:
```
Deprecated config option `[server]` `LFS_CONTENT_PATH` present. Use `[lfs]` `PATH` instead. This fallback will be/has been removed in v1.19.0
```
```
Deprecated config option `[log]` `ROUTER` present. Use `[log]` `logger.router.MODE` instead. This fallback will be/has been removed in 1.21
```
Ideally, it would be nice to use `sed` to fix the config files for users but that's more of a nice to have. I've also added some suggestions above to make the installation wizard screen a bit more compact and cleaner looking.

All in all, the update seems to work well for clean installs as well as updates.

**Comment by wkobiela** [comment]
> Ideally, it would be nice to use `sed` to fix the config files for users but that's more of a nice to have. I've also added some suggestions above to make the installation wizard screen a bit more compact and cleaner looking.

I suppose it would be nice, but hard to guess what custom changes in configuration file particular user made. (I guess that these two deprecated options are just initial setup options - all differences between clear setup of 1.17 vs upgrade to 1.21). 

If it is okay, I would stick with information to update `config.ini` manually

**Review by mreid-tt — APPROVED** [review]

**Comment by mreid-tt** [comment]
LGTM

**Comment by hgy59** [comment]
@mreid-tt can you please publish this?

in my local env, building gitea fails with:
```
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
```

**Comment by mreid-tt** [comment]
@hgy59, the version has been published.

**Pull request merged**